cli/command/system/prune_progress.go

```diff
@@ -0,0 +1,175 @@
+package system
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/docker/cli/cli/command"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/filters"
+	"github.com/docker/docker/pkg/jsonmessage"
+	"github.com/docker/docker/pkg/progress"
+	"github.com/docker/docker/pkg/streamformatter"
+)
+
+type pruneProgressChan struct {
+	isEventsListening chan bool
+	errorChan         chan error
+}
+
+func runPruneProgress(dockerCli command.Cli, progressChans *pruneProgressChan, options pruneOptions) {
+	eventChan := make(chan string, 1)
+	go monitorRemovingEvents(dockerCli, eventChan, progressChans)
+	objCount, err := countObjectsToRemove(dockerCli, options)
+	if err != nil {
+		progressChans.errorChan <- err
+		return
+	}
+	go outputPruningProgress(dockerCli, eventChan, objCount, progressChans)
```

Comment:
minor nit: maybe wrap the `jsonmessage.DisplayJSONMessagesToStream()` call in a `func()` and make that the goroutine instead of `outputPruningProgress` so that both are in the same place